module/os/freebsd/zfs/zvol_os.c

```diff
@@ -772,7 +781,12 @@ zvol_geom_bio_strategy(struct bio *bp)
 	zfs_rangelock_exit(lr);
 
 	bp->bio_completed = bp->bio_length - resid;
-	if (bp->bio_completed < bp->bio_length && off > volsize)
+	if (error == EINTR && bp->bio_completed > 0)
+		error = 0;
+	/* Convert checksum errors into IO errors. */
+	else if (error == ECKSUM)
+		error = SET_ERROR(EIO);
+	if (error == 0 && bp->bio_completed < bp->bio_length && off > volsize)
```

Comment:
Not your's but should it be `off >= size`?
```suggestion
	if (error == 0 && bp->bio_completed < bp->bio_length && off >= volsize)
```